### PR TITLE
chore: try ec2-plugin with `javaPath` option on infra.ci

### PR DESCRIPTION
**Pull request by lemeurherve**
Ref: https://github.com/jenkinsci/ec2-plugin/pull/766

### CODE REVIEW

File: plugins.txt [hunk 1]
```diff
@@ -97,4 +97,4 @@ workflow-job:1232.v5a_4c994312f1
 workflow-multibranch:716.vc692a_e52371b_
 workflow-scm-step:400.v6b_89a_1317c9a_
 workflow-step-api:639.v6eca_cd8c04a_a_
-workflow-support:838.va_3a_087b_4055b
\ No newline at end of file
+workflow-support:838.va_3a_087b_4055b
```
Comment (lemeurherve) [inline plugins.txt:100]: Change of line ending 🤷

**Comment by dduportal** [comment]
The build fail, complaining about missing dependencies.

Make sens since it is an incremental update.

You have to add (temporarly of course) the dependencies listed there: https://plugins.jenkins.io/ec2/#dependencies

**Comment by lemeurherve** [comment]
Meh, I'll pass.

Currently testing it with a manual incremental upload on infra.ci.jenkins.io (with "kubernetes-management" pipeline disabled)

**Comment by lemeurherve** [comment]
Retrieving dependencies names and versions from the plugins site: so much fun...

image

File: plugins.txt [hunk 2]
```diff
@@ -37,7 +37,15 @@ credentials-binding:523.vd859a_4b_122e6
...
-ec2:1.68
+ec2:ec2-1.69-rc1489.404b_49d31b_e7
+aws-java-sdk-ec2:1.12.163-315.v2b_716ec8e4df
+apache-httpcomponents-client-4-api:4.5.13-1.0
+aws-credentials:189.v3551d5642995
+bouncycastle-api:2.25
+node-iterator-api:1.5.1
+trilead-api:1.0.13
+javax-activation-api:1.2.0-2
+javax-mail-api:1.6.2-5
```
Comment (lemeurherve) [inline plugins.txt:48]: Explicit missing dependencies needed for building with the incremental version of the ec2-pliugin

**Comment by lemeurherve** [comment]
Solving dependencies: moar fun...

<details>

```

[2022-08-18T17:25:57.046Z] Plugin azure-container-agents:241.va_780fa_dc374a_ (via jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin azure-container-agents:241.va_780fa_dc374a_ (via jsch:0.1.55.61.va_e9ee26616e7) depends on trilead-api:1.67.vc3938a_35172f, but there is an older version defined on the top level - trilead-api:1.0.13,

[2022-08-18T17:25:57.046Z] Plugin azure-container-agents:241.va_780fa_dc374a_ (via jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin azure-container-agents:241.va_780fa_dc374a_ (via windows-azure-storage:378.vb2cb9797e1e5->copyartifact:1.47) depends on apache-httpcomponents-client-4-api:4.5.13-138.v4e7d9a_7b_a_e61, but there is an older version defined on the top level - apache-httpcomponents-client-4-api:4.5.13-1.0,

[2022-08-18T17:25:57.046Z] Plugin azure-credentials:216.ve0b_4a_485ffc2 (via azure-sdk:118.v43f74dd9ca_dc->jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin azure-credentials:216.ve0b_4a_485ffc2 (via azure-sdk:118.v43f74dd9ca_dc->jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin azure-vm-agents:822.v3a18fc3d2de1 (via jsch:0.1.55.61.va_e9ee26616e7) depends on trilead-api:1.67.vc3938a_35172f, but there is an older version defined on the top level - trilead-api:1.0.13,

[2022-08-18T17:25:57.046Z] Plugin azure-vm-agents:822.v3a18fc3d2de1 (via azure-sdk:118.v43f74dd9ca_dc->jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin azure-vm-agents:822.v3a18fc3d2de1 (via azure-sdk:118.v43f74dd9ca_dc->jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin blueocean-bitbucket-pipeline:1.25.6 (via cloudbees-bitbucket-branch-source:785.ve724eb_44e286) depends on apache-httpcomponents-client-4-api:4.5.13-138.v4e7d9a_7b_a_e61, but there is an older version defined on the top level - apache-httpcomponents-client-4-api:4.5.13-1.0,

[2022-08-18T17:25:57.046Z] Plugin blueocean-bitbucket-pipeline:1.25.6 (via cloudbees-bitbucket-branch-source:785.ve724eb_44e286->jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin blueocean-bitbucket-pipeline:1.25.6 (via cloudbees-bitbucket-branch-source:785.ve724eb_44e286->jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin blueocean-commons:1.25.6 (via jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin blueocean-commons:1.25.6 (via jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin blueocean-github-pipeline:1.25.6 (via jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin blueocean-github-pipeline:1.25.6 (via jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin blueocean-jira:1.25.6 (via jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin blueocean-jira:1.25.6 (via jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin ec2:ec2-1.69-rc1489.404b_49d31b_e7 has unresolvable dependencies: Unable to resolve dependencies for ec2,

[2022-08-18T17:25:57.046Z] Plugin aws-java-sdk-ec2:1.12.163-315.v2b_716ec8e4df (via aws-java-sdk-minimal:1.12.246-349.v96b_b_f7eb_a_c3c->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin aws-java-sdk-ec2:1.12.163-315.v2b_716ec8e4df (via aws-java-sdk-minimal:1.12.246-349.v96b_b_f7eb_a_c3c->jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin git-client:3.11.2 (via jsch:0.1.55.61.va_e9ee26616e7) depends on trilead-api:1.67.vc3938a_35172f, but there is an older version defined on the top level - trilead-api:1.0.13,

[2022-08-18T17:25:57.046Z] Plugin git-forensics:1.9.0 (via forensics-api:1.15.1->data-tables-api:1.12.1-2->jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin git-forensics:1.9.0 (via forensics-api:1.15.1->data-tables-api:1.12.1-2->jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin github:1.35.0 (via github-api:1.303-400.v35c2d8258028->jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin github:1.35.0 (via github-api:1.303-400.v35c2d8258028->jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin github-autostatus:3.6.2 (via cobertura:1.17->code-coverage-api:3.0.5->jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin github-autostatus:3.6.2 (via cobertura:1.17->code-coverage-api:3.0.5->jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin github-branch-source:1677.v731f745ea_0cf (via jjwt-api:0.11.5-77.v646c772fddb_0->jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin github-branch-source:1677.v731f745ea_0cf (via jjwt-api:0.11.5-77.v646c772fddb_0->jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin github-checks:1.0.18 (via github-api:1.303-400.v35c2d8258028->jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin github-checks:1.0.18 (via github-api:1.303-400.v35c2d8258028->jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.046Z] Plugin jira:3.7.1 (via jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.047Z] Plugin jira:3.7.1 (via jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.047Z] Plugin junit:1119.1121.vc43d0fc45561 (via jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.047Z] Plugin junit:1119.1121.vc43d0fc45561 (via jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.047Z] Plugin kubernetes:3697.v771155683e38 (via jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.047Z] Plugin kubernetes:3697.v771155683e38 (via jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.047Z] Plugin kubernetes-credentials:0.9.0 (via jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.047Z] Plugin kubernetes-credentials:0.9.0 (via jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.047Z] Plugin kubernetes-credentials-provider:1.196.va_55f5e31e3c2 (via kubernetes-client-api:5.12.2-193.v26a_6078f65a_9->jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.047Z] Plugin kubernetes-credentials-provider:1.196.va_55f5e31e3c2 (via kubernetes-client-api:5.12.2-193.v26a_6078f65a_9->jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.047Z] Plugin metrics:4.2.10-389.v93143621b_050 (via jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.047Z] Plugin metrics:4.2.10-389.v93143621b_050 (via jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.047Z] Plugin pipeline-model-api:2.2114.v2654ca_721309 (via jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.047Z] Plugin pipeline-model-api:2.2114.v2654ca_721309 (via jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.047Z] Plugin pipeline-rest-api:2.24 (via jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.047Z] Plugin pipeline-rest-api:2.24 (via jackson2-api:2.13.3-285.vc03c0256d517->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.047Z] Plugin ssh-credentials:295.vced876c18eb_4 depends on trilead-api:1.67.vc3938a_35172f, but there is an older version defined on the top level - trilead-api:1.0.13,

[2022-08-18T17:25:57.047Z] Plugin warnings-ng:9.18.0 depends on apache-httpcomponents-client-4-api:4.5.13-138.v4e7d9a_7b_a_e61, but there is an older version defined on the top level - apache-httpcomponents-client-4-api:4.5.13-1.0,

[2022-08-18T17:25:57.047Z] Plugin warnings-ng:9.18.0 (via jackson2-api:2.13.3-285.vc03c0256d517) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2,

[2022-08-18T17:25:57.047Z] Plugin warnings-ng:9.18.0 (via analysis-model-api:10.16.0->jaxb:2.3.6-1) depends on javax-activation-api:1.2.0-3, but there is an older version defined on the top level - javax-activation-api:1.2.0-2

```

</details>

**Comment by lemeurherve** [comment]
Self-dependency now? Come on...

```
[2022-08-18T17:37:41.497Z] Plugin ec2:ec2-1.69-rc1489.404b_49d31b_e7 has unresolvable dependencies: Unable to resolve dependencies for ec2
 	at io.jenkins.tools.pluginmanager.impl.PluginManager.start(PluginManager.java:240)
 	at io.jenkins.tools.pluginmanager.impl.PluginManager.start(PluginManager.java:189)
 	at io.jenkins.tools.pluginmanager.cli.Main.main(Main.java:52)
 	Suppressed: io.jenkins.tools.pluginmanager.impl.PluginDependencyException: Plugin ec2:ec2-1.69-rc1489.404b_49d31b_e7 has unresolvable dependencies: Unable to resolve dependencies for ec2
 		at io.jenkins.tools.pluginmanager.impl.PluginManager.resolveRecursiveDependencies(PluginManager.java:1102)
 		at io.jenkins.tools.pluginmanager.impl.PluginManager.findPluginsAndDependencies(PluginManager.java:692)
 		at io.jenkins.tools.pluginmanager.impl.PluginManager.start(PluginManager.java:232)
 		... 2 more
 	Caused by: io.jenkins.tools.pluginmanager.impl.DownloadPluginException: Unable to resolve dependencies for ec2
 		at io.jenkins.tools.pluginmanager.impl.PluginManager.resolveDependenciesFromManifest(PluginManager.java:946)
 		at io.jenkins.tools.pluginmanager.impl.PluginManager.resolveDirectDependencies(PluginManager.java:1052)
 		at io.jenkins.tools.pluginmanager.impl.PluginManager.resolveRecursiveDependencies(PluginManager.java:1098)
 		... 4 more
```

File: plugins.txt [hunk 3]
```diff
@@ -37,7 +37,15 @@ credentials-binding:523.vd859a_4b_122e6
 dark-theme:185.v276b_5a_8966a_e
 datadog:4.0.0
 design-library:104.v492ed3cce3f2
-ec2:1.68
+ec2:incrementals;org.jenkins-ci.plugins;ec2-1.69-rc1489.404b_49d31b_e7;lemeurherve;JENKINS-69304-java-path
```
Comment (dwnusbaum) [inline plugins.txt:40]: Maybe try dropping the `ec2` prefix from the incremental version to match the `workflow-support` examples in https://github.com/jenkinsci/incrementals-tools#updating-versions-for-jenkins-docker-images.

```suggestion
ec2:incrementals;org.jenkins-ci.plugins;1.69-rc1489.404b_49d31b_e7;lemeurherve;JENKINS-69304-java-path
```

**Comment by lemeurherve** [comment]
> Maybe try dropping the `ec2` prefix from the incremental version to match the examples in [jenkinsci/incrementals-tools#updating-versions-for-jenkins-docker-images](https://github.com/jenkinsci/incrementals-tools#updating-versions-for-jenkins-docker-images).

That was it, thanks again @dwnusbaum!

**Pull request merged**